code/modules/clothing/suits/storage.dm

```diff
@@ -69,18 +69,27 @@
 
 	if(open == 1) //Will check whether icon state is currently set to the "open" or "closed" state and switch it around with a message to the user
 		open = 0
-		icon_state = initial(icon_state)
+		update_icon()
 		flags_inv = HIDETIE|HIDEHOLSTER
 		to_chat(usr, "You button up the coat.")
 	else if(open == 0)
 		open = 1
-		icon_state = "[icon_state]_open"
+		update_icon()
 		flags_inv = HIDEHOLSTER
 		to_chat(usr, "You unbutton the coat.")
 	else //in case some goofy admin switches icon states around without switching the icon_open or icon_closed
 		to_chat(usr, "You attempt to button-up the velcro on your [src], before promptly realising how silly you are.")
 		return
```

Comment:
```suggestion
	if(open < 0)//in case some goofy admin switches icon states around without switching the icon_open or icon_closed
		to_chat(usr, "You attempt to button-up the velcro on your [src], before promptly realising how silly you are.")
		return
	open = !open
	flags_inv = open ? HIDETIE : HIDETIE|HIDEHOLSTER
	update_icon()
	to_chat(usr, "You [open ? "un" : ""]button \the [src].")